### PR TITLE
chore: replace SwiftLint build phase by aggregate target - WPB-8588

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8588" title="WPB-8588" target="_blank">EpicWPB-8588</a>  Modernising code base - Migrate Objective-C to Swift
  </td></table>
  <br />
 

### Issue

Running SwiftLint increases the build time quite much.
This PR creates an Aggregate target which allows for running SwiftLint manually instead.

### Testing

Open Xcode, choose the "Lint Swift Code" scheme and build.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios/Wire-iOS.xcodeproj/xcshareddata/xcschemes/Format Swift Code (SwiftFormat).xcscheme
Comment (netbe) [inline wire-ios/Wire-iOS.xcodeproj/xcshareddata/xcschemes/Format Swift Code (SwiftFormat).xcscheme:1]: I would suggest to do one aggregated target that lint and format code

**Review by netbe — DISMISSED** [review]
Let's remove also the precommit swiftlint

File: .swiftlint.yml [hunk 1]
```diff
@@ -107,4 +107,4 @@ line_length:
   warning: 500
   error: 500
 
-swiftlint_version: 0.54.0
+swiftlint_version: 0.55.1
```
Comment (markusfassbender) [inline .swiftlint.yml:110]: This doubles the work @agisilaos  and me did already on some branch recently (I lost track which one). But thanks you updated everything also according to the new rules, we just disabled it 😁

**Review by markusfassbender — APPROVED** [review]

**Review by agisilaos — APPROVED** [review]

File: WireDomain/Sources/WireDomain/Repositories/UpdateEventsRepository.swift [hunk 2]
```diff
@@ -122,9 +122,8 @@ final class UpdateEventsRepository: UpdateEventsRepositoryProtocol {
         try await eventContext.perform { [eventContext, encoder] in
             let data = try encoder.encode(eventEnvelope)
 
-            if let string = String(data: data, encoding: .utf8) {
-                print("persisting event: \(string)")
-            }
+            let string = String(decoding: data, as: UTF8.self)
```
Comment (samwyndham) [inline WireDomain/Sources/WireDomain/Repositories/UpdateEventsRepository.swift:125]: Interesting. Never, seen this initializer before! Will it always succeed to create some kind of string from random data  or does it crash? 

This comment isn't a request to change anything. I'm just interested in this API.

**Review by netbe — APPROVED** [review]

File: wire-ios-link-preview/WireLinkPreview/MetaStreamContainer.swift [hunk 3]
```diff
@@ -58,8 +58,6 @@ final class MetaStreamContainer {
     }
 
     private func parseString(from data: Data) -> String? {
-        return String(data: data, encoding: .utf8)
-            ?? String(data: data, encoding: .isoLatin1)
```
Comment (samwyndham) [inline wire-ios-link-preview/WireLinkPreview/MetaStreamContainer.swift:62]: Just a sanity check that it is OK to get rid of `String(data: data, encoding: .isoLatin1)` here?

File: wire-ios-link-preview/WireLinkPreview/MetaStreamContainer.swift [hunk 4]
```diff
@@ -58,8 +58,6 @@ final class MetaStreamContainer {
     }
 
     private func parseString(from data: Data) -> String? {
```
Comment (samwyndham) [inline wire-ios-link-preview/WireLinkPreview/MetaStreamContainer.swift:60]: I think it is worth making this non optional now for some additional cleanup.

**Pull request merged**